spec/models/accounting/quickbooks/transaction_creator_spec.rb

```diff
@@ -6,24 +6,126 @@
   let(:class_service) { instance_double(Quickbooks::Service::Class, find_by: [class_ref]) }
   let(:connection) { instance_double(Accounting::Quickbooks::Connection) }
   let(:account) { instance_double(Accounting::Account, qb_id: 98) }
-  subject { described_class.new(instance_double(Division, qb_connection: connection, principal_account: account)) }
+  let(:creator) { described_class.new(instance_double(Division, qb_connection: connection, principal_account: account)) }
 
   before do
-    allow(subject).to receive(:service).and_return(generic_service)
-    allow(subject).to receive(:class_service).and_return(class_service)
+    allow(generic_service).to receive(:create)
+    allow(creator).to receive(:service).and_return(generic_service)
+    allow(creator).to receive(:class_service).and_return(class_service)
   end
 
-  it 'should create class ref, and create journal entry' do
-    expect(generic_service).to receive(:create)
-    expect(class_service).to receive(:find_by)
-
-    subject.add_disbursement(
-      amount: 12.09,
-      loan_id: 2,
-      memo: 'I am memo',
-      description: 'I am a line item description',
-      qb_bank_account_id: 89,
-      qb_customer_id: 3
-    )
+  context 'when qb customer does exist' do
+    let(:qb_customer_id) { 3 }
+    let(:loan_id) { 2 }
+    let(:amount) { 78.20 }
+    let(:memo) { 'I am a memo' }
+    let(:description) { 'I am a line item description' }
+
+    subject do
+      creator.add_disbursement(
+        amount: amount,
+        loan_id: loan_id,
+        memo: memo,
+        description: description,
+        qb_bank_account_id: 89,
+        qb_customer_id: qb_customer_id
+      )
+    end
+
+    context 'generic_service' do
```

Comment:
A complex model is created for the JournalEntry object needed for QBO. These specs are testing that the values are in the correct place. 

I see two potential issues with the code:
1. This code is tightly coupled to the QBO model. I am abusing mocks too much here? Is there value in these kind of specs?
2. There is also quite a bit of duplication. It possibly could be cleaned up with the right pattern.